core/src/test/java/io/el/concurrent/DefaultEventLoopGroupTest.java

```diff
@@ -0,0 +1,279 @@
+      if (inEventLoop()) {
+        throw new IllegalStateException("cannot await termination of the current thread");
+      }
+
+      try {
+        threadLock.await(timeout, unit);
+      } catch (InterruptedException e) {
+      }
+      return isTerminated();
+    }
+
+    @Override
+    public boolean isTerminated() {
+      return state.equals(State.TERMINATED);
+    }
+  }
+
+  private static class MockDefaultEventLoopGroup extends DefaultEventLoopGroup {
+
+    protected MockDefaultEventLoopGroup(int nThreads, Executor executor,
+        EventLoopChooserFactory chooserFactory) {
+      super(nThreads, executor, chooserFactory);
+    }
+
+    @Override
+    public void shutdown() {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Runnable> shutdownNow() {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> tasks)
+        throws InterruptedException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> tasks, long timeout,
+        TimeUnit unit) throws InterruptedException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> tasks)
+        throws InterruptedException, ExecutionException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> tasks, long timeout, TimeUnit unit)
+        throws InterruptedException, ExecutionException, TimeoutException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    protected EventLoop newChild(Executor executor) {
+      return eventLoop;
+    }
+
+    @Override
+    protected ThreadFactory newDefaultThreadFactory() {
+      return null;
+    }
+  }
+
+  private static class MockFailedDefaultEventLoopGroup extends MockDefaultEventLoopGroup {
+
+    protected MockFailedDefaultEventLoopGroup(int nThreads, Executor executor,
+        EventLoopChooserFactory chooserFactory) {
+      super(nThreads, executor, chooserFactory);
+    }
+
+    @Override
+    protected EventLoop newChild(Executor executor) {
+      try {
+        throw new Exception();
+      } catch (Exception e) {
+
+      }
+      return null;
+    }
+  }
+
+  @Nested
+  @DisplayName("On DefaultEventLoopGroup constructor")
+  class DefaultEventLoopGroupConstructor {
+
+    @Test
+    @DisplayName("When construct with proper newChild() method, then shutdownGracefully and shutdown")
+    void testConstructorSucceed() {
+      try {
+        new MockFailedDefaultEventLoopGroup(
+            3,
+            mockExecutor,
+            eventLoopChooserFactory
+        );
+      } catch (IllegalStateException e) {
+        assertFalse(defaultEventLoopGroup.isTerminated());
+      }
+    }
```

Comment:
Is this a success case?

Reply:
Yes. I ll change the displayName contents.